include/random_walks/hamiltonian_monte_carlo_walk.hpp

```diff
@@ -125,7 +129,7 @@ struct HamiltonianMonteCarloWalk {
       solver->set_state(1, v);
 
       // Get proposals
-      solver->steps(walk_length);
+      solver->steps(walk_length, accepted);
```

Comment:
Does this line suffice to perform `walk_length` leapfrog steps?

Reply:
Yes. We should also add thinning (=keep sample every M steps of the sampler each of which runs for walk_length steps).